### PR TITLE
feature/mouseMovement

### PR DESCRIPTION
**Pull request by EnriquePuyol**
Now you can move the Scene camera by holding the middle click of the mouse and moving it

### CODE REVIEW

File: Source/ModuleCamera.cpp [hunk 1]
```diff
@@ -129,7 +136,7 @@ void ModuleCamera::InputRotate(float dt) const
 	if (App->input->GetMouseButtonDown(SDL_BUTTON_RIGHT) == KEY_REPEAT)
 	{
 		float mouse_motion_x, mouse_motion_y;
-		App->input->GetMouseMotion(mouse_motion_x, mouse_motion_y);
+		App->input->SetMouseMotion(mouse_motion_x, mouse_motion_y);
```
Comment (AlbertVVila) [inline Source/ModuleCamera.cpp]: Here you are not setting the mouse motion but obtaining it in the mouse_motion variables passed by reference

File: Source/ModuleCamera.cpp [hunk 2]
```diff
@@ -150,7 +157,7 @@ void ModuleCamera::InputOrbit(float dt) const
 	if (App->input->IsKeyPressed(SDL_SCANCODE_LALT) && App->input->GetMouseButtonDown(SDL_BUTTON_LEFT) == KEY_REPEAT)
 	{
 		float mouse_motion_x, mouse_motion_y;
-		App->input->GetMouseMotion(mouse_motion_x, mouse_motion_y);
+		App->input->SetMouseMotion(mouse_motion_x, mouse_motion_y);
```
Comment (AlbertVVila) [inline Source/ModuleCamera.cpp]: Same

File: Source/ModuleCamera.cpp [hunk 3]
```diff
@@ -119,6 +119,13 @@ void ModuleCamera::InputMove(float dt) const
 		{
 			movement *= 2;
 		}
+		editorcamera->Move(movement*dt*App->renderer->current_scale);
+	}
+	else if (App->input->GetMouseButtonDown(SDL_BUTTON_MIDDLE) == KEY_REPEAT)
+	{
+		float3 motion = { App->input->GetMouseMotion().x, App->input->GetMouseMotion().y, 0 };
+
+		float3 movement = (-motion.x * editorcamera->frustum->WorldRight() * 1000) + (motion.y * float3::unitY * 1000);
```
Comment (AlbertVVila) [inline Source/ModuleCamera.cpp]: Try avoiding magic numbers

File: Source/ModuleCamera.cpp (hunk 1 above)
Comment (EnriquePuyol) [reply Source/ModuleCamera.cpp]: Yeah but the final result is that you are setting a variable, a Getter cannot be a void function
Apart from that I need a real Getter function to make the mouse movement as you can see in the ModuleCamera.cpp -> line 126

File: Source/ModuleCamera.cpp (hunk 3 above)
Comment (EnriquePuyol) [reply Source/ModuleCamera.cpp]: Will fix that

**Comment by AniolAndres** [comment]
Camera does not move accordingly to the plane displayed, try looking from above  and moving up and down

File: Source/ModuleCamera.cpp (hunk 3 above)
Comment (AniolAndres) [inline Source/ModuleCamera.cpp]: should use frustumUp vector instead of unitY

File: Source/ModuleCamera.cpp (hunk 1 above)
Comment (AlbertVVila) [reply Source/ModuleCamera.cpp]: I know this is merged but a setter if a function that modifies a value. This setter is not modifying anything, it's only getting the mouse motion values and putting them into mouse_motion_x and mouse_motion_y. I think it should be GetMouseMotion.

**Pull request merged**